rs/nns/governance/src/gen/ic_nns_governance.pb.v1.rs

```diff
@@ -128,6 +128,12 @@ pub struct NeuronInfo {
     /// after the UNIX epoch).
     #[prost(uint64, optional, tag = "13")]
     pub voting_power_refreshed_timestamp_seconds: ::core::option::Option<u64>,
+    /// See the analogous field in Nueron.
+    #[prost(uint64, optional, tag = "14")]
+    pub deciding_voting_power: ::core::option::Option<u64>,
```

Comment:
These fields don't really belong in the internal representation of the protobuf here.  These types can still be used for storing neurons (and we may migrate away from the segmented storage).  

These may belong on the API fields, but I don't think they should be on the internal here.

Is it possible to rework this to only afffect the API types?

Reply:
I don't think this is ideal, but I am not sure how to avoid this in a decent way. The problem is that in order to populate these, it is not enough to have Neuron. You also need the time (and in the future, you will also need VotingPowerEconomics 😖). If you have **_concrete_** suggestions for how to achieve your goals, please, **_show me_**. Otherwise, I do not know how to act on this feedback.

Reply:
I don't know yet what concrete suggestions to make, because I haven't really wrestled with it yet.

what options did you consider and reject before doing it this way?  What can we come up with?  If we agree that it's not ideal, what would it take to move it back in a better direction?

Reply:
Maybe we can brainstorm and come up with a good design.... it will likely be relevant for other endpoints in the near future too.

Reply:
My thoughts were in the same direction, but with a bit of a difference.

I don't know that we should change any of the base level conversions.  We really only need this field to be populated when we're populating the API response, no?   The Neuron conversions in every other case are perhaps just mechanical.  

That's to say, we don't need to replace the From implementation, we only need to use the into_api() method in places where we are preparing a response.  

I think limiting the scope in that way makes it more manageable.  

Further, we might consider this an opportunity to look at the design of the canister.rs and the main Governance methods that are used in it.   The main problem seems to be that when we are preparing the response, the fields wouldn't be available on the non-API types, which would make it tricky to do the conversions in the canister.rs where we no longer have access to the original Neuron (i.e. non-proto neuron).  

That seems to be the place where some new design idea is needed.   Would you agree with that?

Reply:
> I don't know that we should change any of the base level conversions. We really only need this field to be populated when we're populating the API response, no?

Not sure what you mean by "base level conversions"...

My main requirement is that governance.did stays as-is. If you know a good way to do that while changing governance.proto back to the way it was, I an open to that.

> That's to say, we don't need to replace the From implementation, we only need to use the into_api() method in places where we are preparing a response.

I think being able to convert from proto::Neuron(Info) to api::Neuron(Info) without populating these new fields would be a foot gun, particularly since you can **_implicitly_** call `from` via `.into()` (which I thought we agree to not do anymore). Moreover, I do not see when you would ever actually want to not populate these fields. AFAICT, the only reason you are converting to api::Neuron(Info) is that you are about to send out a response.

Therefore, I really think we would want to get rid of `from` and replace it with `into_api` or something like this.

> Would you agree with that?

You mean, how do api types calculate these new fields without native Neuron? Yes, it is not super clear how to do that. I would think that you would simply copy `fn deciding_voting_power` and `fn potential_voting_power` from native Neuron to api::Neuron.